src/test/java/org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojoTest.java

```diff
@@ -54,17 +66,34 @@ public void testDefaultConfig() throws Exception {
         }
     }
 
-    public void testInvalidFormatWithSkipExec() throws Exception {
-        File pluginXmlFile = new File(getBasedir(), "src/test/resources/plugin-configs/check-plugin-config.xml");
-
-        Mojo mojo = lookupMojo("check", pluginXmlFile);
-
-        assertNotNull("Mojo not found.", mojo);
+    @Provides
+    public MavenProjectStub projectStub() {
+        return new MavenProjectStub() {
+            public File getFile() {
+                return new File(getBasedir(), "target/classes");
+            }
 
```

Comment:
MojoExtension provide a mock for MavenProject to check if we need create stub

Can be as cleanup in next PR

Reply:
Thx. I clean it up.